# E-commerce Sales Analysis



In [1]:
import pandas as pd

df = pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")
df.head()


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [2]:
## Basic Information
print("Rows and Columns:", df.shape)
print("\nColumn Names:")
print(df.columns.tolist())

print("\nDataset Information:")
df.info()


Rows and Columns: (100, 13)

Column Names:
['Order_ID', 'Order_Date', 'Customer', 'City', 'Region', 'Category', 'Product', 'Quantity', 'Unit_Price', 'Discount_Percent', 'Total_Sales', 'Rating', 'Payment_Method']

Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    object 
 2   Customer          100 non-null    object 
 3   City              100 non-null    object 
 4   Region            100 non-null    object 
 5   Category          100 non-null    object 
 6   Product           100 non-null    object 
 7   Quantity          100 non-null    int64  
 8   Unit_Price        100 non-null    int64  
 9   Discount_Percent  100 non-null    int64  
 10  Total_Sales       100 non-null    float64
 11  Rating            100 non-null    float64
 12 

In [3]:
# Selecting one column
df["Product"].head()


,Product
0,Wireless Headphones
1,Running Shoes
2,Power Bank
3,Mechanical Keyboard
4,Smart Watch


In [4]:
# Selecting multiple columns
df[["Product", "Category", "Quantity", "Total_Sales"]].head()


,Product,Category,Quantity,Total_Sales
0,Wireless Headphones,Electronics,3,4497.00
1,Running Shoes,Sports,1,2799.00
2,Power Bank,Electronics,2,2038.30
3,Mechanical Keyboard,Electronics,4,9496.20
4,Smart Watch,Electronics,1,3134.05


### Filtering Data


In [5]:
# Orders where sales are more than 5000
df[df["Total_Sales"] > 5000].head(10)


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card
10,1011,2026-03-22,Rahul Das,Hyderabad,South,Home & Kitchen,Coffee Maker,2,3499,0,6998.00,4.2,UPI
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
13,1014,2026-02-05,Dev Patel,Jaipur,North,Sports,Running Shoes,3,2799,0,8397.00,4.6,UPI
16,1017,2026-03-17,Sana Malik,Hyderabad,South,Sports,Cricket Bat,3,2499,0,7497.00,4.0,Net Banking
18,1019,2026-05-19,Reyansh Jain,Chennai,South,Clothing,Jacket,3,2499,0,7497.00,4.7,UPI
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card


In [6]:
# Orders from the Electronics category
df[df["Category"] == "Electronics"].head()


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
14,1015,2026-06-24,Sana Malik,Kolkata,East,Electronics,Wireless Headphones,2,1499,0,2998.00,4.8,UPI


In [7]:
# Orders where quantity is 3 or more
df[df["Quantity"] >= 3].head()


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.0,3.8,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.2,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.2,4.3,Credit Card
8,1009,2026-06-05,Aman Kumar,Mumbai,West,Home & Kitchen,Electric Kettle,3,1499,0,4497.0,4.0,Cash on Delivery
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.4,4.7,Credit Card


## Sorting Data

In [8]:
# Highest sales first
df.sort_values("Total_Sales", ascending=False).head(10)


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
47,1048,2026-06-24,Sara Ahmed,Lucknow,North,Home & Kitchen,Coffee Maker,3,3499,0,10497.00,5.0,UPI
69,1070,2026-06-30,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,3.8,Net Banking
79,1080,2026-05-25,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,4.6,Debit Card
51,1052,2026-03-13,Rohan Mehta,Chandigarh,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.2,Debit Card
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card


In [9]:
# Lowest sales first
df.sort_values("Total_Sales", ascending=True).head(10)


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
62,1063,2026-04-17,Aman Kumar,Chandigarh,North,Books,The Alchemist,1,499,10,449.10,4.8,Credit Card
73,1074,2026-03-28,Alina Shah,Lucknow,North,Books,Atomic Habits,1,599,10,539.10,5.0,Net Banking
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
60,1061,2026-05-12,Alina Shah,Bengaluru,South,Books,Python Programming,1,749,5,711.55,5.0,Cash on Delivery
82,1083,2026-06-25,Priya Menon,Jammu,North,Sports,Football,1,799,5,759.05,3.8,Cash on Delivery
30,1031,2026-01-04,Rohan Mehta,Ahmedabad,West,Sports,Yoga Mat,1,899,15,764.15,4.1,Credit Card
42,1043,2026-05-22,Kabir Ali,Jaipur,North,Clothing,T-Shirt,1,799,0,799.00,4.2,Debit Card
20,1021,2026-01-18,Reyansh Jain,Chandigarh,North,Sports,Yoga Mat,1,899,10,809.10,4.7,Credit Card
77,1078,2026-01-05,Maryam Khan,Kolkata,East,Books,Machine Learning Basics,1,999,10,899.10,4.7,UPI


##  Basic Sales Statistics

In [10]:
total_sales = df["Total_Sales"].sum()
average_sales = df["Total_Sales"].mean()
highest_sales = df["Total_Sales"].max()
lowest_sales = df["Total_Sales"].min()
total_quantity = df["Quantity"].sum()
number_of_orders = df["Order_ID"].nunique()

print("Total Sales:", total_sales)
print("Average Sales:", round(average_sales, 2))
print("Highest Sale:", highest_sales)
print("Lowest Sale:", lowest_sales)
print("Total Quantity Sold:", total_quantity)
print("Number of Orders:", number_of_orders)


Total Sales: 447262.95000000007
Average Sales: 4472.63
Highest Sale: 13995.0
Lowest Sale: 449.1
Total Quantity Sold: 294
Number of Orders: 100


##  Category-wise Sales Analysis

In [12]:
category_sales = df.groupby("Category")["Total_Sales"].sum()
category_sales.sort_values(ascending=False)


,Total_Sales
Category,
Sports,136026.45
Home & Kitchen,113329.90
Electronics,92086.00
Clothing,69028.20
Books,36792.40


In [11]:
# Category-wise quantity sold
category_quantity = df.groupby("Category")["Quantity"].sum()
category_quantity.sort_values(ascending=False)


,Quantity
Category,
Sports,79
Books,60
Home & Kitchen,59
Electronics,57
Clothing,39


In [13]:
# Category-wise average sales
df.groupby("Category")["Total_Sales"].mean().sort_values(ascending=False)


,Total_Sales
Category,
Sports,5441.058000
Home & Kitchen,5396.661905
Clothing,4930.585714
Electronics,4846.631579
Books,1752.019048


## City-wise Sales Analysis

In [15]:
city_sales = df.groupby("City")["Total_Sales"].sum()
city_sales.sort_values(ascending=False)


,Total_Sales
City,
Hyderabad,59076.45
Chennai,47493.20
Lucknow,46194.65
Pune,41908.55
Mumbai,41457.80
Kochi,37257.50
Chandigarh,34539.65
Delhi,30882.80
Jaipur,25534.00


In [14]:
# Cities with the highest number of orders
df.groupby("City")["Order_ID"].nunique().sort_values(ascending=False)


,Order_ID
City,
Chennai,10
Pune,10
Kochi,9
Chandigarh,8
Mumbai,8
Jaipur,7
Hyderabad,7
Kolkata,7
Jammu,6


##  Product-wise Sales Analysis

In [16]:
product_sales = df.groupby("Product")["Total_Sales"].sum()
product_sales.sort_values(ascending=False).head(10)


,Total_Sales
Product,
Coffee Maker,51610.25
Jacket,47106.15
Cricket Bat,41483.40
Running Shoes,41285.25
Wireless Headphones,33877.40
Dumbbell Set,30184.90
Mechanical Keyboard,26489.40
Electric Kettle,26382.40
Mixer Grinder,25341.55


In [17]:
# Top products by quantity sold
df.groupby("Product")["Quantity"].sum().sort_values(ascending=False).head(10)


,Quantity
Product,
Wireless Headphones,23
Atomic Habits,21
Jacket,20
Electric Kettle,19
Cricket Bat,18
The Alchemist,17
Coffee Maker,16
Dumbbell Set,16
Yoga Mat,16


##  Payment Method Analysis

In [18]:
payment_sales = df.groupby("Payment_Method")["Total_Sales"].sum()
payment_sales.sort_values(ascending=False)


,Total_Sales
Payment_Method,
Credit Card,106540.25
Debit Card,98064.90
Net Banking,96371.25
UPI,96163.25
Cash on Delivery,50123.30


In [19]:
# Number of orders for each payment method
df.groupby("Payment_Method")["Order_ID"].nunique().sort_values(ascending=False)


,Order_ID
Payment_Method,
UPI,26
Credit Card,22
Debit Card,20
Net Banking,17
Cash on Delivery,15


## Finding a Specific Order

In [20]:
# Order with the highest sales
df[df["Total_Sales"] == df["Total_Sales"].max()]


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.0,4.0,Net Banking


In [21]:
# Total sales, average sales and quantity for each category
df.groupby("Category").agg({
    "Total_Sales": ["sum", "mean", "max", "min"],
    "Quantity": "sum"
}).sort_values(("Total_Sales", "sum"), ascending=False)


Total_Sales                                 Quantity
                       sum         mean       max      min      sum
Category                                                           
Sports           136026.45  5441.058000  13995.00   759.05       79
Home & Kitchen   113329.90  5396.661905  12596.40  1188.30       59
Electronics       92086.00  4846.631579   9496.20  1708.10       57
Clothing          69028.20  4930.585714  10620.75   679.15       39
Books             36792.40  1752.019048   4495.00   449.10       60

## 13. Observations

After analysing the dataset, I found:

- The dataset contains **100 orders** and the total quantity sold is **294 items**.
- Total sales are about **₹4,47,262.95**, while the average sale per order is about **₹4,472.63**.
- **Sports** is the highest-selling category with sales of about **₹1,36,026.45**.
- **Hyderabad** has the highest total sales among the cities, with about **₹59,076.45**.
- **Coffee Maker** is the top product by total sales, with about **₹51,610.25**.
- **Credit Card** generated the highest sales among the payment methods, with about **₹1,06,540.25**.
- The highest-value order was **Order ID 1049**, for **Running Shoes**, with total sales of **₹13,995**.
- The lowest-value order was **Order ID 1006**, for **The Alchemist**, with total sales of **₹449.10**.

